pkg/kubelet/volumes.go

```diff
@@ -143,7 +143,8 @@ func (kl *Kubelet) mountExternalVolumes(pod *api.Pod) (kubecontainer.VolumeMap,
 			return nil, err
 		}
 		if attacher != nil {
-			err = attacher.Attach(volSpec, kl.hostname)
+			host := &volumeHost{kl}
+			err = attacher.Attach(host, volSpec, kl.hostname)
```

Comment:
> I think the challenge there is that the path of the device is determined by the call to AttachDisk not from the disk id as is the case with GCE

The `waitForAttach` call for the AWS EBS plugin will have to issue another call to AWS to figure out if the volume is attached, and if so where it is mounted. AWS [`describe-volumes`](http://docs.aws.amazon.com/cli/latest/reference/ec2/describe-volumes.html), for example, returns the state (e.g. `"State": "attached"`) and the mount path (e.g. `"Device": "/dev/sda1"`.

> I would say open a PR with the work done for AWS EBS and base that on another PR which has all the changes to the Attacher/Detacher interface that you need and let's carry on this discussion there. Otherwise we are trying to hit a moving target.

Agreed, further discussion here is unwarranted. We can discuss on the AWS EBS PR. I'd argue that no further changes (other than https://github.com/kubernetes/kubernetes/pull/25523) are needed to the Attacher/Detacher interface.